internal/dag/httpproxy_processor.go

```diff
@@ -1473,6 +1474,21 @@ func (p *HTTPProxyProcessor) GlobalAuthorizationContext() map[string]string {
 	return nil
 }
 
+// getSortedHttpVersions returns and empty slice or ["h2", "http/1.1"] or ["http/1.1"].
+// This is done to conform with how envoy expects AlpnProtocols in tlsv3.CommonTlsContext.
+func (p *HTTPProxyProcessor) getSortedHttpVersions(proxy *contour_api_v1.HTTPProxy) []string {
```

Comment:
It'd be nice to reduce the duplication between this new method, `cmd/contour/parseDefaultHTTPVersions` and `internal/envoy/v3/ProtoNamesForVersions`. I'll have to look some more for specific ideas on how to refactor for reuse but if you have ideas in the meantime they are welcome.

Reply:
Do agree to move `parseDefaultHTTPVersions` and `ProtoNamesForVersions` into a util package?